src/plugins/vis_type_table_new/public/components/table_vis_component.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { orderBy } from 'lodash';
+import { EuiDataGridProps, EuiDataGrid, EuiDataGridSorting, EuiTitle } from '@elastic/eui';
+
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { Table } from '../table_vis_response_handler';
+import { TableVisConfig, ColumnWidth, SortColumn, TableUiState } from '../types';
+import { getDataGridColumns } from './table_vis_grid_columns';
+import { usePagination } from '../utils';
+import { convertToFormattedData } from '../utils/convert_to_formatted_data';
+import { TableVisControl } from './table_vis_control';
+
+interface TableVisComponentProps {
+  title?: string;
+  table: Table;
+  visConfig: TableVisConfig;
+  event: IInterpreterRenderHandlers['event'];
+  uiState: TableUiState;
+}
+
+export const TableVisComponent = ({
+  title,
+  table,
+  visConfig,
+  event,
+  uiState,
+}: TableVisComponentProps) => {
+  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
+    table,
+    visConfig
+  );
+
+  const pagination = usePagination(visConfig, rows.length);
+
+  const sortedRows = useMemo(() => {
+    return uiState.sort && uiState.sort.colIndex !== null && uiState.sort.direction
```

Comment:
nit - could simplify to 
```suggestion
    return uiState.sort?.colIndex && uiState.sort.direction
```

Reply:
will change